include/nbl/video/IPhysicalDevice.h

```diff
@@ -581,33 +619,15 @@ class NBL_API2 IPhysicalDevice : public core::Interface, public core::Unmovable
             );
         }
 
-        template<typename FORMAT_USAGE>
-        struct FormatPromotionRequest
-        {
+        struct SBufferFormatPromotionRequest {
             asset::E_FORMAT originalFormat = asset::EF_UNKNOWN;
-            FORMAT_USAGE usages = FORMAT_USAGE();
-
-            struct hash
-            {
-                // pack into 64bit for easy hashing 
-                uint64_t operator()(const FormatPromotionRequest<FORMAT_USAGE>& r) const
-                {
-                    uint64_t msb = uint64_t(std::hash<FORMAT_USAGE>()(r.usages));
-                    return (msb << 32u) | r.originalFormat;
-                }
-            };
-
-            struct equal_to
-            {
-                bool operator()(const FormatPromotionRequest<FORMAT_USAGE>& l, const FormatPromotionRequest<FORMAT_USAGE>& r) const
-                {
-                    return l.originalFormat == r.originalFormat && l.usages == r.usages;
-                }
-            };
+            detail::SFormatBufferUsages::SUsage usages = detail::SFormatBufferUsages::SUsage();
         };
 
-        using SBufferFormatPromotionRequest = FormatPromotionRequest<IPhysicalDevice::SFormatBufferUsages::SUsage>;
-        using SImageFormatPromotionRequest = FormatPromotionRequest<IPhysicalDevice::SFormatImageUsages::SUsage>;
+        struct SImageFormatPromotionRequest {
+            asset::E_FORMAT originalFormat = asset::EF_UNKNOWN;
+            detail::SFormatImageUsages::SUsage usages = detail::SFormatImageUsages::SUsage();
+        };
```

Comment:
this doesn't look at nice as the previous code but removing the template seems to workaround the crash for clang ...